pkg/controller/trainjob_controller.go

```diff
@@ -266,6 +270,18 @@ func setTerminalCondition(ctx context.Context, runtime jobruntimes.Runtime, trai
 	return nil
 }
 
+func setJobsStatus(ctx context.Context, runtime jobruntimes.Runtime, trainJob *trainer.TrainJob) error {
+	statuses, err := runtime.JobsStatus(ctx, trainJob)
+	if err != nil {
+		return err
+	}
+	if statuses == nil {
+		return nil
+	}
+	trainJob.Status.JobsStatus = statuses
+	return nil
```

Comment:
Do we need this check for nil status ?
```suggestion
	trainJob.Status.JobsStatus = statuses
	return nil
```